lib/kredis/types/proxy.rb

```diff
@@ -9,19 +9,19 @@ def initialize(redis, key, **options)
     options.each { |key, value| send("#{key}=", value) }
   end
 
-  def multi(&block)
-    # NOTE: to be removed when Redis 4 compatibility gets dropped
-    return redis.multi unless block
-
-    redis.multi do |pipeline|
-      block.call(Kredis::Types::Proxy.new(pipeline, key))
+  def multi(*args, **kwargs, &block)
+    redis.multi(*args, **kwargs) do |pipeline|
+      Thread.current[:pipeline] = pipeline
+      block.call
+    ensure
+      Thread.current[:pipeline] = nil
```

Comment:
Consider using a `thread_mattr_accessor` so that it takes care of using a key based on the attribute name and the class name. If not, I'd try to make `:pipeline` more specific: e.g: :kredis_pipeline or something like that.